### PR TITLE
[patch] memory leak bugfix unclosed error channel on http handler

### PR DESCRIPTION
**Pull request by kpango**
unclosed channel makes memory leak
Signed-off-by: kpango <i.can.feel.gravity@gmail.com>

### CODE REVIEW

File: router/router.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ func routing(m []string, t time.Duration, h handler.Func) http.Handler {
 				go func() {
 					// it is the responsibility for handler to close the request
 					ech <- h(w, r.WithContext(ctx))
+					close(ech)
```
Comment (WindzCUHK) [inline router/router.go]: seems this is not the root cause, as GC will handle it
although it is better to close it explicitly...


Else, client sidecar should have the same problem
https://github.com/yahoojapan/athenz-client-sidecar/blob/841c41e48cb336b1eba6223a6b470bc55e7fe130/router/router.go#L60-L63

Comment (kpango) [reply router/router.go]: yes, unclosed channel will collect by GC, but it uses a lot of memory until GC, that is why our memory metrics were spiked.

if you want check unclosed channel please run blelow code
```go
package main

func main() {
    for i := 0; i < 10000000; i++ {
        c := make(chan int)
        defer close(c)
        go func(){
            select {
            case <-c:
                return
            }
        }()
    }
}
```

Comment (WindzCUHK) [reply router/router.go]: I think the go func cannot return and release reference, as the channel is blocked.
GC seems not running in this case...

```go
package main

import "time"

func main() {
	// var c chan int
	for i := 0; i < 10000000; i++ {
		c := make(chan int)
		// defer close(c) // 1

		go func() {
			// defer close(c) // 2
			select {
			case <-c:
				return
			default:
				return
			}
		}()
	}

	time.Sleep(1 * time.Hour)
}
```

1. not closing, memory ~= 6.7 MB
1. close in position 1, memory ~= 1.5 GB !!!
1. close in position 2, memory ~= 7.2 MB

Seems the problem is cause by slacking defer() in the main goroutine...

Comment (kpango) [reply router/router.go]: no your code is wrong, if you use default statement, grouting will not block,
there is no gc wait

**Comment by kevindiu** [comment]
LGTM.

**Pull request merged**